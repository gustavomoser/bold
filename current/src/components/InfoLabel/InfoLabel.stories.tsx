import { text } from '@storybook/addon-knobs'
import React from 'react'

import { VFlow } from '../VFlow'

import { InfoLabel } from './InfoLabel'

export default {
  title: 'Components/InfoLabel',
}

export const Default = () => (
  <VFlow vSpacing={0.5}>
    <InfoLabel title={text('title', 'Name')} placeholder={text('placeholder', '-')}>
      Jo√£o da Silva
    </InfoLabel>
    <InfoLabel title={text('title', 'Text')} placeholder={text('placeholder', '-')}>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Explicabo quasi ullam eos, fuga earum molestiae cum
      odit, modi exercitationem quis hic quae. Excepturi quia neque expedita explicabo facere corporis dolore?
    </InfoLabel>
    <InfoLabel title={text('title', 'Name')} placeholder={text('placeholder', '-')} />
  </VFlow>
)
