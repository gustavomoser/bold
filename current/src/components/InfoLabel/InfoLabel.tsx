import React, { CSSProperties } from 'react'

import { Theme, useStyles } from '../../styles'

export interface InfoLabelProps {
  title: string
  titleStyles?: any
  childStyles?: any
  placeholder?: string
  children?: React.ReactNode
}

export function InfoLabel(props: InfoLabelProps) {
  const { title, children, titleStyles, childStyles } = props
  const { classes, css } = useStyles(createStyles, props)

  return (
    <div className={classes.infoLabel}>
      <div className={css(classes.title, titleStyles)}>{title}</div>
      <div className={css(classes.child, childStyles)}>{children}</div>
    </div>
  )
}

export const createStyles = (theme: Theme, { placeholder }: InfoLabelProps) => ({
  infoLabel: {
    lineHeight: 1.5,
  } as CSSProperties,
  title: {
    fontWeight: 'bold',
  } as CSSProperties,
  child: {
    ':empty::before': {
      content: `"${placeholder || '-'}"`,
      color: theme.pallete.text.secondary,
      fontStyle: 'italic',
    },
  } as CSSProperties,
})
