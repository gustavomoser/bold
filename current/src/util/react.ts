import { MutableRefObject, Ref } from 'react'

/**
 * Compose multiple react refs.
 *
 * @returns A new ref that passes down the Element to all ref arguments.
 */
export function composeRefs<T>(...refs: Array<Ref<T>>) {
  return (value: T) => refs.forEach(ref => setRef(ref, value))
}

/**
 * Changes the value of a ref.
 * The ref can be a function ref or a object ref.
 *
 * @param ref The ref to have its value changed.
 * @param value The new value.
 */
export function setRef<T>(ref: Ref<T>, value: any) {
  if (typeof ref === 'function') {
    ref(value)
  } else if (ref) {
    const mutableRef = ref as MutableRefObject<T>
    mutableRef.current = value
  }
}

/**
 * Compose multiple event handlers.
 *
 * @returns A new function that triggers all handlers passed as parameters, in order.
 */
export function composeHandlers(...handlers: Array<(...params: any) => void>) {
  return (...args: any[]) => handlers.forEach(handler => handler && handler(...args))
}

/**
 * Invokes the native `value` property setter of an element.
 * From https://github.com/facebook/react/issues/10135#issuecomment-401496776
 *
 * @param element The element to invoke the setter on.
 * @param value The value to be set.
 */
export function setNativeValue(element: HTMLElement, value: any) {
  const prototype = Object.getPrototypeOf(element)
  const { set: prototypeValueSetter = null } = Object.getOwnPropertyDescriptor(prototype, 'value') || {}
  const { set: valueSetter = null } = Object.getOwnPropertyDescriptor(element, 'value') || {}

  if (prototypeValueSetter && valueSetter !== prototypeValueSetter) {
    prototypeValueSetter.call(element, value)
  } else if (valueSetter) {
    valueSetter.call(element, value)
  } else {
    throw new Error('The given element does not have a value setter')
  }
}
