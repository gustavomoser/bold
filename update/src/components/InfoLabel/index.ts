export { InfoLabel } from './InfoLabel'
export type { InfoLabelProps } from './InfoLabel'
