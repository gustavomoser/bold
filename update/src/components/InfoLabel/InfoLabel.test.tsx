import { render } from '@testing-library/react'
import React from 'react'

import { InfoLabel } from './InfoLabel'

it('should render correctly', () => {
  expect(render(<InfoLabel title='Test'>Content</InfoLabel>).container).toMatchSnapshot()

  expect(
    render(
      <InfoLabel title='Test' titleStyles={{ color: 'red' }} childStyles={{ color: 'blue' }}>
        Content
      </InfoLabel>
    ).container
  ).toMatchSnapshot()
})

it('should accept placeholder', () => {
  expect(render(<InfoLabel title='Test' placeholder='No value' />).container).toMatchSnapshot()
})
