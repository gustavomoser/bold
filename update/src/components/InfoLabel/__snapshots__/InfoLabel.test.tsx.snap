// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept placeholder 1`] = `
.emotion-2 {
  line-height: 1.5;
}

.emotion-0 {
  font-weight: bold;
}

.emotion-1:empty::before {
  content: "No value";
  color: #696979;
  font-style: italic;
}

<div>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    >
      Test
    </div>
    <div
      class="emotion-1"
    />
  </div>
</div>
`;

exports[`should render correctly 1`] = `
.emotion-2 {
  line-height: 1.5;
}

.emotion-0 {
  font-weight: bold;
}

.emotion-1:empty::before {
  content: "-";
  color: #696979;
  font-style: italic;
}

<div>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    >
      Test
    </div>
    <div
      class="emotion-1"
    >
      Content
    </div>
  </div>
</div>
`;

exports[`should render correctly 2`] = `
.emotion-2 {
  line-height: 1.5;
}

.emotion-0 {
  font-weight: bold;
  color: red;
}

.emotion-1 {
  color: blue;
}

.emotion-1:empty::before {
  content: "-";
  color: #696979;
  font-style: italic;
}

<div>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    >
      Test
    </div>
    <div
      class="emotion-1"
    >
      Content
    </div>
  </div>
</div>
`;
